### ISSUE TITLE
bad u-turn on road with turn restriction

### ISSUE THREAD

**Issue by RedAuburn** [graphhopper/graphhopper#2992]
**Describe the bug**
routing tries to make a u-turn to get to a road that has a turn restriction

**To Reproduce**
https://www.openstreetmap.org/directions?engine=graphhopper_car&route=51.47523%2C0.01003%3B51.47462%2C0.00983

**Expected behaviour**
should be routed the fastest legal way:
https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=51.47523%2C0.01003%3B51.47462%2C0.00983

**Screenshots**

current / correct

![image1](https://github.com/graphhopper/graphhopper/assets/26939824/f7152274-6c1e-48da-8244-32daf06b1c19)

**Comment by AmyMagdalena** [comment]
Similar issue here, biking route makes illegal turn, rather than telling cyclists to dismount and use the pedestrian crossing, 

Road Tags: 
```
bicycle=yes
highway=tertiary
horse=no
lanes=2
maxspeed=30
motor_vehicle=yes
name=Carrer Gran de Gràcia
name:ca=Carrer Gran de Gràcia
name:etymology:wikidata=Q852697
oneway=yes
restriction=no_left_turn
surface=asphalt
wikidata=Q1805479
wikipedia=ca:Carrer Gran de Gràcia
```


https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=41.40466%2C2.15116%3B41.40537%2C2.15000#map=19/41.40539/2.15064
**
![imatge](https://github.com/graphhopper/graphhopper/assets/98350291/ac799fc6-5a5a-41aa-aa8f-9ab0a56665dd)
**

**Comment by AmyMagdalena** [comment]
Yes, I fixed that and am aware the cache takes time to update, I just don't think it'll ever choose to dismount over making the illegal turn. As you say, it's not considered.
